devel/query.lua

```diff
@@ -620,6 +631,22 @@ function printField(path, field, value)
     end
 end
 
+function print_linked_list(name, next)
+    if string.find(name, "next") then
+        local JobListItem=next --danke F.F.
+        while(JobListItem.next) do
```

Comment:
You might find `for _, item in utils.listpairs()` useful here.

Reply:
Certainly. I've taken a crack at revising the `foreach(table, name, callback(field, value))` function to include it as a third pairs option.

```
elseif string.find(name,".*list") and table["next"] then
```
I use this check to decide on using that pairs function, but if there is a more precise and elegant way for this too I would gladly use it.

The formatting I've so far decided to use for this particular type of data is borrowing repetitions from regex
```lua
local m = tostring(field):gsub("<.*: ",""):gsub(">.*",""):gsub("0x", "0x "):gsub("%x%x%x%x%x%x","%1 ",1)
local s = string.format("next{%d}.item [next: %s]", index, m)
-- which outputs as lines like this
| next{34}.item [next: 0x7fff6f 3b5440]: ~  <job: 0x7fff6d86ca80> [struct]
```
All the options I thought made sense are here
```
string.format("*{%d}next.item", index), "[next.item]{%d}")
string.format("*[%d]next.item", index), "**next.item"
string.format("next.item{%d}", index), "*{r}.next.item"
```